### ISSUE TITLE
[Problem/Bug]: BrowserProcess crashes with exit code -2147483645 when canceling the FrameNavigationStarting event (for a particular site)

### ISSUE THREAD

**Issue by pushkin-** [MicrosoftEdge/WebView2Feedback#4843]
### What happened?

When canceling the FrameNavigationStarting event of certain sites, the browser process crashes. Exit code: -2147483645

![{121715B2-ADD8-4CEA-B971-6A3915EC826D}](https://github.com/user-attachments/assets/d404a4f3-f47a-431c-b9d2-08eae7126b7d)

### Importance

Important. My app's user experience is significantly compromised.

### Runtime Channel

Stable release (WebView2 Runtime)

### Runtime Version

129.0.2792.65

### SDK Version

1.0.2535.41

### Framework

Winforms

### Operating System

Windows 11

### OS Version

22631.4169

### Repro steps

modify the WinForms sample webview2 app as follows:
1. Iin `WebView2Control_CoreWebView2InitializationCompleted`, add: `this.webView2Control.CoreWebView2.NewWindowRequested += CoreWebView2_NewWindowRequested;`
2. make that handler create the webview2 itself and add a FrameNavigationStarting handler:
```
	private async void CoreWebView2_NewWindowRequested(object sender, CoreWebView2NewWindowRequestedEventArgs e)
	{
		e.Handled = true;
		using (e.GetDeferral())
		{
			Form form = new Form();
			WebView2 webView = new WebView2();
			await webView.EnsureCoreWebView2Async();
			form.Controls.Add(webView);

			webView.CoreWebView2.FrameNavigationStarting += CoreWebView2_FrameNavigationStarting;
			form.Visible = true;
			webView.Dock = DockStyle.Fill;
			e.NewWindow = webView.CoreWebView2;
		}
	}
```
3. set the FrameNavigationStarting handler to:
```
	private void CoreWebView2_FrameNavigationStarting(object sender, CoreWebView2NavigationStartingEventArgs e)
	{
		if (e.Uri.StartsWith("https://smartscreentestratings.com/xplt_100_0_0.html"))
		{
			e.Cancel = true;
		}
	}
```

1. start app and navigate to https://demo.smartscreen.msft.net/
2. scroll down and click the Malvertising link (which will open a window to https://demo.smartscreen.msft.net/other/exploit_frame.html)
3. observe browser process crashes

### Repros in Edge Browser

No, issue does not reproduce in the corresponding Edge version

### Regression

No, this never worked

[AB#54319772](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/54319772)

**Comment by LiangTheDev** [comment]
Thanks for reporting the issue. I've filed an internal work item to track and investigate.

**Comment by dhveerap** [comment]
A fixed has been made, it will be available in Edge Runtime version >= 132.0.2947.0
